include/gzip/decompress.hpp

```diff
@@ -13,84 +13,73 @@ namespace gzip {
 class Decompressor
 {
     std::size_t max_;
+    struct libdeflate_decompressor* decompressor_ = nullptr;
```

Comment:
👋 @GretaCB great question.

First context:

 - The libdeflate API requires you allocate a compressor struct pointer per https://github.com/ebiggers/libdeflate/blob/9a327aae416a71471015b70f4a10f8b10cf1db31/libdeflate.h#L40-L55
 - That pointer can either be null or valid
 - If libdeflate were written in C++ it would likely use a `std::optional` wrapper (or some other way to handle the "could create this thing" vs "could not create this thing" cases.
 - But because libdeflate is written in C it uses this (fairly common in my experience) way of modeling an object that might be valid: being either null or not.
 - A key design principle of `gzip-hpp` is to wrap compression functionality it an idiomatic C++ API (that is hard to use incorrectly and therefore much more secure). So here we need to wrap the libdeflate logic in a C++ class.

Now my answer:

 - I chose to have the `decompressor_` set to null here since:
  - that feels like the right default state for it to be in (until the class is created)
  - for best performance we only want to have to initialize the decompressor once